### PR TITLE
Add configuration of new repository and default branch protection

### PR DESCRIPTION
**Pull request by felddy**
#  #

## 🗣 Description ##

This PR adds the ability to correctly setup a new repository's configuration options and default branch protection as specified in the development guide:
 - https://github.com/cisagov/development-guide/blob/develop/project_setup/branch-protection.md
 - https://github.com/cisagov/development-guide/tree/develop/project_setup#disabling-squash-merging

Note that this PR is moving parallel to #1.  

## 💭 Motivation and context ##

These GitHub API calls were used when bringing the myriad repos back public during the great darkness of 2021.
As we found out... when repositories are made private and then returned public, branch protection settings are lost.

This change will further reduce the number of manual steps needed to get a new repository into shape.

This also captures this special sauce so we can use it to remediate any configuration deviations in the future.  These changes  might be useful as a separate GitHub extension that just updates the repo config and branch protection.

## 🧪 Testing ##

This was tested on repositories in both `cisagov` and my personal account by creating new repositories from an existing skeleton.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]
Looks good to me, sir!

File: gh-skeleton [hunk 1]
```diff
@@ -182,14 +182,108 @@ Please manually create the ${dest_org}/${dest_repo} repository on GitHub.
...
+  branch=$3
+
+  log_info "Configuring repository options for ${dest_org}/${dest_repo}."
+
+  jq -n "{
+      allow_auto_merge: false,
+      allow_merge_commit: true,
+      allow_rebase_merge: true,
+      allow_squash_merge: false,
+      default_branch: \"$3\",
```
Comment (dav3r) [inline gh-skeleton]: I'm not sure if the curly braces should be here or not, but I am pretty sure the `$3` should be `$branch`, no?
```suggestion
      default_branch: \"${branch}\",
```

**Review by dav3r — CHANGES_REQUESTED** [review]
I see one minor thang that should be corrected here.

File: gh-skeleton (hunk 1 above)
Comment (felddy) [reply gh-skeleton]: Good catch.  

With regards to the braces... it doesn't matter either way for this usage.  Here is a good explanation of when braces are helpful/required:  

https://stackoverflow.com/questions/8748831/when-do-we-need-curly-braces-around-shell-variables

**Review by dav3r — APPROVED** [review]
Excellent sauce! 🌶️

**Pull request merged**